test/gpu/jit.cpp

```diff
@@ -206,8 +206,16 @@ TEST_CASE(compile_warnings)
     EXPECT(not compile("").empty());
     EXPECT(not compile("-Wunused-parameter -Wno-error").empty());
     EXPECT(not compile("-Wno-unused-parameter -Werror").empty());
+#ifdef MIGRAPHX_USE_HIPRTC
+    if(not migraphx::enabled(migraphx::gpu::MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS{}))
+    {
+        EXPECT(test::throws([&] { compile("-Werror=unused-parameter"); }));
+        EXPECT(test::throws([&] { compile("-Wunused-parameter -Werror"); }));
+    }
+#else
     EXPECT(test::throws([&] { compile("-Werror=unused-parameter"); }));
     EXPECT(test::throws([&] { compile("-Wunused-parameter -Werror"); }));
+#endif
```

Comment:
It would be better to just remove the ifdef and the just do:

```cpp
    if(not migraphx::enabled(migraphx::gpu::MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS{}))
    {
        EXPECT(test::throws([&] { compile("-Werror=unused-parameter"); }));
        EXPECT(test::throws([&] { compile("-Wunused-parameter -Werror"); }));
    }
``` 

I dont think anyone will set `MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS` without using hiprtc in the first place.

Reply:
> I dont think anyone will set MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS without using hiprtc in the first place. 

It could happen if someone is switching between hipclang and hiprtc locally. They could have have `MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS` env set in their env while using hip-clang and that would disable this test.